gdx/src/com/badlogic/gdx/maps/tiled/renderers/BatchTiledMapRenderer.java

```diff
@@ -186,7 +186,22 @@ public void renderObject (MapObject object) {
 	@Override
 	public void renderImageLayer (TiledMapImageLayer layer) {
 		final Color batchColor = batch.getColor();
-		final float color = Color.toFloatBits(batchColor.r, batchColor.g, batchColor.b, batchColor.a * layer.getOpacity());
+		final Color combinedTint = layer.getCombinedTintColor();
+
+		// Check if layer supports transparency
+		boolean supportsTransparency = layer.supportsTransparency();
+
+		// If the Image Layer supports transparency we do not want to modify the combined tint during rendering
+		// and if the Image Layer does not support transparency, we want to multiply the combined tint values, by its alpha
+		float alphaMultiplier = supportsTransparency ? 1f : combinedTint.a;
+		// Only modify opacity by combinedTint.b if Image Layer supports transparency
+		float opacityMultiplier = supportsTransparency ? combinedTint.a : 1f;
+
+		// For image layer rendering multiply all by alpha
+		// except for opacity when image layer does not support transparency
+		final float color = Color.toFloatBits(batchColor.r * (combinedTint.r * alphaMultiplier),
+			batchColor.g * (combinedTint.g * alphaMultiplier), batchColor.b * (combinedTint.b * alphaMultiplier),
+			batchColor.a * (layer.getOpacity() * opacityMultiplier));
```

Comment:
If I see it right, the same code is also in the other renderers. Maybe a new utility function would be nice to not have much code duplication

Reply:
Yep, you are totally correct. A lot of the tiledmap rendering code is pretty messy. Each individual map type has it's own separate rendering methods for TileLayers and ImageLayers, with a lot of duplicated code paired with minor differences within.
When I first started working on this long ago. I didn't want to rock the boat and make any big changes or refactoring, adding utility classes, etc. For just simple color tinting feature request. So I tried my best to just go along with how it was done at the time.

That being said I will now gladly take another look at this and pair it down. 

Thanks!